### ISSUE TITLE
Screen refresh rate

### ISSUE THREAD

**Issue by capezoloh** [adamples/VU_meter#1]
Neat little project you got here. Most of these kind of projects I see on youtube have really poor screen update rate. Is this true for yours aswell?

**Comment by adamples** [comment]
Yeah, I also noticed it. There are several reasons for this. Most projects use u8g2, which is a great general-purpose library, but with some limitations arising from it's design goals. In the picture loop mode it refreshes the whole screen, which takes considerable amount of time just to transfer the data (~24ms for I2C at 400kHz). Additionally it uses synchronous I2C driver, so it can't render while sending the data either. This yields FPS in range of 20+.

In this project I aimed at different approach: asynchronous I2C, using HW available in atmega, and partial screen updates. It also uses sprites instead of drawing primitives directly to the buffer. It all requires additional data and flash memory, but in the end you get update time of ~7ms, so more than 60FPS for 2 displays (it's more than display's actual frame rate I believe). It gives an impression of really smooth movement.

The code was tailored for purpose of displaying the needle against fixed background. It can be changed to do picture-loop style drawing or to use framebuffer, and calculating an update can be generalized too, but it's not the purpose of this project.
